eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -62,50 +62,35 @@ function create-metadata-table($readmeFolder, $readmeName, $moniker, $msService,
   $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
     -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
     -msService $msService
-  Add-Content -Path $readmePath -Value $metadataString
+  Add-Content -Path $readmePath -Value $metadataString -NoNewline
 
   # Add tables, seperate client and mgmt.
   $readmeHeader = "# Azure $serviceName SDK for $languageDisplayName - $moniker`r`n"
   Add-Content -Path $readmePath -Value $readmeHeader
-  Add-Content -Path $readmePath -Value $content
-}
-
-function compare-and-merge-metadata ($original, $updated) {
-  if (!$original -or !$updated) {
-    return $original + $updated 
-  }
-  $originalTable = ConvertFrom-StringData -StringData $original -Delimiter ":"
-  $updatedTable = ConvertFrom-StringData -StringData $updated -Delimiter ":"
-  foreach ($key in $originalTable.Keys) {
-    if (!($updatedTable.ContainsKey($key))) {
-      Write-Warning "New metadata missed the entry: $key. Adding back."
-      $updated += "$key`: $($originalTable[$key])`r`n"
-    }
-  }
-  return $updated
+  Add-Content -Path $readmePath -Value $content -NoNewline
 }
 
 # Update the metadata table.
 function update-metadata-table($readmeFolder, $readmeName, $serviceName, $msService)
 {
   $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
   $readmeContent = Get-Content -Path $readmePath -Raw
-  # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
-  $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
-    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
-    -msService $msService
   $match = $readmeContent -match "^---\n*(?<metadata>(.*\n?)*?)---\n*(?<content>(.*\n?)*)"
   if (!$match) {
-    Set-Content -Path $readmePath -Value "$metadataString`r`n$readmeContent" -NoNewline
+    # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
+    $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
```

Comment:
Why duplicate this call? I would suggest if there is no match that you just pass in original metadata as null instead of duplicating the contents of this very long line. Inside your function you can decide if there is anything to merge or not.

Reply:
If it doesn't match, then the line below throw error instead of null. We need to use try catch to make it correct

<img width="385" alt="image" src="https://user-images.githubusercontent.com/48036328/178796176-90368390-51b4-4ec0-b47f-39bd0cd58049.png">

Reply:
$Matches["metadata"] is the statement throw errors.